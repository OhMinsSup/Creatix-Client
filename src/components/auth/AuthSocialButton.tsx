import * as React from 'react';
import styled, { css } from 'styled-components';
import {
  FaFacebookF as FacebookIcon,
  FaGoogle as GoogleIcon,
  FaGithub as GithubIcon,
} from 'react-icons/fa';
import palette from '../../lib/styles/palette';

const AuthSocialButtonBlock = styled.button<{ border: boolean }>`
  width: 3rem;
  height: 3rem;
  border-radius: 1.5rem;
  display: flex;
  color: white;
  align-items: center;
  justify-content: center;
  outline: none;
  border: none;
  cursor: pointer;
  transition: 0.125s all ease-in;
  ${props =>
    props.border &&
    css`
      border: 1px solid ${palette.gray3};
    `}
  &:focus {
    box-shadow: 0px 2px 12px rgba(0, 0, 0, 0.35);
  }
`;

export interface AuthSocialButtonProps {
  provider: 'facebook' | 'google' | 'github';
  tabIndex?: number;
}

const providerMap = {
  github: {
    color: '#272e33',
    icon: GithubIcon,
    border: false,
  },
  google: {
    color: '#f03e3e',
    icon: GoogleIcon,
    border: true,
  },
  facebook: {
    color: '#3b5998',
    icon: FacebookIcon,
    border: false,
  },
};

const AuthSocialButton: React.SFC<AuthSocialButtonProps> = ({
  provider,
  tabIndex,
}) => {
  const info = providerMap[provider];
  const { icon: Icon, color, border } = info;
  return (
    <AuthSocialButtonBlock
      style={{
        background: color,
      }}
      border={border}
      tabIndex={tabIndex}
    >
      <Icon />
    </AuthSocialButtonBlock>
  );
};

export default AuthSocialButton;
