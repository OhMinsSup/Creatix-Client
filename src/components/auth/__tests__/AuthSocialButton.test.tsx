import React from 'react';
import { render } from 'react-testing-library';
import AuthSocialButton, { AuthSocialButtonProps } from '../AuthSocialButton';

describe('AuthSocialButton', () => {
  const setup = (props: Partial<AuthSocialButtonProps> = {}) => {
    const initialProps: AuthSocialButtonProps = {
      provider: 'facebook',
      tabIndex: 4,
    };
    return render(<AuthSocialButton {...initialProps} {...props} />);
  };

  it('renders properly', () => {
    setup();
  });

  it('renders correctly', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('gives facebook background style to AuthSocialButton', () => {
    const { container } = setup({
      provider: 'facebook',
    });

    const facebook = container.querySelector('button');
    expect(facebook && facebook.style.background).toEqual('rgb(59, 89, 152)');
  });

  it('gives google background style to AuthSocialButton', () => {
    const { container } = setup({
      provider: 'google',
    });

    const google = container.querySelector('button');
    expect(google && google.style.background).toEqual('rgb(240, 62, 62)');
  });

  it('gives github background style to AuthSocialButton', () => {
    const { container } = setup({
      provider: 'github',
    });

    const github = container.querySelector('button');
    expect(github && github.style.background).toEqual('rgb(39, 46, 51)');
  });
});
