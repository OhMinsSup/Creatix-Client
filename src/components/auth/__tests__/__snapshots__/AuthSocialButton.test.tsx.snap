// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthSocialButton renders correctly 1`] = `
.c0 {
  width: 3rem;
  height: 3rem;
  border-radius: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: white;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  border: none;
  cursor: pointer;
  -webkit-transition: 0.125s all ease-in;
  transition: 0.125s all ease-in;
}

.c0:focus {
  box-shadow: 0px 2px 12px rgba(0,0,0,0.35);
}

<div>
  <button
    class="c0"
    style="background: rgb(59, 89, 152);"
    tabindex="4"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 264 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M76.7 512V283H0v-91h76.7v-71.7C76.7 42.4 124.3 0 193.8 0c33.3 0 61.9 2.5 70.2 3.6V85h-48.2c-37.8 0-45.1 18-45.1 44.3V192H256l-11.7 91h-73.6v229"
      />
    </svg>
  </button>
</div>
`;
